pkg/logging/logger.go

```diff
@@ -73,7 +84,18 @@ func UpdateConfig(logLevel string, logFormat string, logFile string) error {
 			w = file
 		}
 	}
-	Log.SetOutput(w)
-	Log.Debug("logging configured")
+	Log.Logger.SetOutput(w)
+	Log.Logger.Debug("logging configured")
 	return nil
 }
+
+// RegisterLogger a custom logger
+func RegisterLogger(l *logrus.Entry) {
+	Log = l
+	customLoggerRegistered = true
```

Comment:
Do we need to call `metadata.AddMetadataToLogger()` after we swap out the logger, or does this happen early enough that we haven't run it in `PersistentPreRun` on the root command yet?